src/pip/_internal/utils/subprocess.py

```diff
@@ -189,9 +191,10 @@ def call_subprocess(
         )
         proc.stdin.close()
     except Exception as exc:
-        subprocess_logger.critical(
-            "Error %s while executing command %s", exc, command_desc,
-        )
+        if log_failed_cmd:
```

Comment:
I would leave `call_subprocess` as it was. Given that the existing behavior (with git at least) is to show the errors if e.g. the command is missing, doing the same with hg should be fine. It's not ideal, but I would rather do this right by refactoring `call_subprocess` in a followup instead of making its behavior more complicated for yet another caller.

Reply:
In that case this is OK, IMO. It's not *too* much more difficult to reason about, and at least it will capture our intention so when this is refactored it will be taken into account (vs having a separate issue).